tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -206,6 +210,27 @@ def forward(self, x):
         my_algo.train(datasamples=(x_train, y_train), _skip=True)
 
 
+@pytest.mark.parametrize(
+    "n_samples,y_shape",
+    [(5, 3), (5, 1), (10, 2), (1, 1), (1, 10)],
```

Comment:
In general, we try to limit the parametrization in order to limit the number of tests to whats really needed :)